src/main/java/seedu/address/model/appointment/AppointmentListView.java

```diff
@@ -0,0 +1,185 @@
+package seedu.address.model.appointment;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.appointment.exceptions.AppointmentNotFoundException;
+import seedu.address.model.appointment.exceptions.DuplicateAppointmentException;
+import seedu.address.model.person.Name;
+
+
+/**
+ * A list of appointments that enforces uniqueness between its elements and does not allow nulls.
+ * An appointment is considered unique by comparing using {@code Appointment#isSameAppointment(Appointment)}.
+ * As such, adding and updating of appointments uses Appointment#isSameAppointment(Appointment) for equality
+ * to ensure that the appointment being added or updated is unique and not duplicated.
+ * However, the removal of an appointment uses Person#equals(Object) to ensure that the
+ * appointment with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see AppointmentView#isSameAppointmentView(AppointmentView)
+ */
+public class AppointmentListView implements Iterable<AppointmentView> {
+
+    private final ObservableList<AppointmentView> internalList = FXCollections.observableArrayList();
+    private final ObservableList<AppointmentView> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent appointment as the given argument.
+     */
+    public boolean contains(AppointmentView toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameAppointmentView);
+    }
+
+    /**
+     * Adds an appointment to the list.
+     * The appointment must not already exist in the list.
+     */
+    public void add(AppointmentView toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateAppointmentException();
+        }
+        internalList.add(toAdd);
+        sortList();
+    }
+
+    /**
+     * Replaces the appointment {@code target} in the list with {@code editedAppointment}.
+     * {@code target} must exist in the list.
+     * The appointment of {@code editedAppointment} must not be the same as another existing appointment in the list.
+     */
+    public void setAppointment(AppointmentView target, AppointmentView editedAppointment) {
```

Comment:
Maybe the Java Docs and function names can be updated to match `AppointmentView` instead of `Appointment`